### PR TITLE
Added method AddValue to Variable

### PR DESCRIPTION
**Pull request by orenbm**
- Added method AddValue to Variable
- parsing variable id with Uri.EscapeDataString as WebUtility.UrlEncode does not comply with the server
- Using System.Json for AddValue content
- Added test for AddValue

### CODE REVIEW

File: test/VariablesTest.cs [hunk 1]
```diff
@@ -11,15 +11,24 @@ public VariablesTest()
...
         }
+
+        [Test]
+        public void TestAddValue()
+        {
+            Mocker.Mock (new Uri ("test:///variables/foo%2Fbar/values"), "testvalue");
+            Client.Variable ("foo/bar").AddValue ("testvalue");
+
+            string res = Mocker.Mock (new Uri ("test:///variables/foo%2Fbar/value"), "testvalue").Read ();
+            Assert.AreEqual (res, "testvalue");
```
Comment (jvanderhoof) [inline test/VariablesTest.cs]: What's actually happening here? Are we just creating a mock object to return a set string and verifying the string matches? If so, it feels like we remove any real testing value by mocking out the response.

File: test/VariablesTest.cs [hunk 2]
```diff
@@ -11,15 +11,24 @@ public VariablesTest()
...
         [Test]
-        public void TestValue()
+        public void TestGetValue()
         {
             Mocker.Mock(new Uri("test:///variables/foo%2Fbar/value"), "testvalue");
             Assert.AreEqual("testvalue", Client.Variable("foo/bar").GetValue());
 
             // TODO: not sure if this is supposed to be a plus or %20 or either
-            Mocker.Mock(new Uri("test:///variables/foo+bar/value"), "space test");
+            Mocker.Mock(new Uri("test:///variables/foo%20bar/value"), "space test");
```
Comment (dividedmind) [inline test/VariablesTest.cs]: Looks like you checked and the plus doesn't work. if so, please remove the comment above.

File: conjur-api/Variable.cs [hunk 3]
```diff
@@ -38,5 +39,25 @@ public string GetValue()
         {
             return this.Client.AuthenticatedRequest(this.path + "/value").Read();
         }
+
+        /// <summary>
+        /// Adds a value to the variable.
+        /// </summary>
```
Comment (dividedmind) [inline conjur-api/Variable.cs]: I know the documentation convention we've adopted here is verbose bordering on the ridiculous, but please for the sake of consistency add the `<param>` here.

File: conjur-api/Variable.cs [hunk 4]
```diff
@@ -38,5 +39,25 @@ public string GetValue()
...
+        public void AddValue (string value)
+        {
+            var req = this.Client.AuthenticatedRequest ($"{this.path}/values");
```
Comment (dividedmind) [inline conjur-api/Variable.cs]: Please keep the formatting consistent. In particular please don't use a space before the method parameters.

File: conjur-api/Variable.cs [hunk 5]
```diff
@@ -27,7 +28,7 @@ public class Variable : Resource
         internal Variable(Client client, string name)
             : base(client, "variable", name)
         {
-            this.path = "variables/" + WebUtility.UrlEncode(name);
+            this.path = "variables/" + Uri.EscapeDataString(name);
```
Comment (dividedmind) [inline conjur-api/Variable.cs]: Good catch! :)

File: test/VariablesTest.cs (hunk 1 above)
Comment (dividedmind) [reply test/VariablesTest.cs]: I agree, this test does not appear to actually test anything.

File: test/WebMocker.cs [hunk 6]
```diff
@@ -46,6 +47,15 @@ public override WebHeaderCollection Headers
                 set;
             }
 
+            public override string ContentType {
+                get {
```
Comment (dividedmind) [inline test/WebMocker.cs]: Please move braces to the next line to keep formatting consistent.

File: test/VariablesTest.cs (hunk 1 above)
Comment (dividedmind) [reply test/VariablesTest.cs]: Note how it still passes if you remove `AddValue` implementation.

File: conjur-api/Variable.cs (hunk 3 above)
Comment (orenbm) [reply conjur-api/Variable.cs]: Done

File: conjur-api/Variable.cs (hunk 4 above)
Comment (orenbm) [reply conjur-api/Variable.cs]: Oh good catch! That's probably the default format in Mac's VS.

**Comment by dividedmind** [comment]
Merged in 8809ac45470f4c6058a4ff48cc138f41787409ad

**Pull request closed without merging**